src/ring_theory/localization.lean

```diff
@@ -365,6 +387,99 @@ end
 lemma of.injective : function.injective (of : β → fraction_ring β) :=
 (is_add_group_hom.injective_iff _).mpr eq_zero_of
 
+section map
+open function is_ring_hom
+variables {A : Type u} [integral_domain A] [decidable_eq A]
+variables {B : Type v} [integral_domain B] [decidable_eq B]
+variables (f : A → B) [is_ring_hom f]
+
+def map (hf : injective f) : fraction_ring A → fraction_ring B :=
+quotient.lift (λ rs : A × (non_zero_divisors A), (f rs.1 : fraction_ring B) / (f rs.2.val)) $
```

Comment:
Can this be defined using `localization.rec`. Also maybe there should be a version of `rec` (which should be called lift I think) for fraction rings if it ends up being easier to use, and a generalization of `map` to localizations, but maybe this has a very ugly statement

Reply:
Rant with tongue in cheek:

There is currently an obfuscated machine called `localization.rec`. However, it is complicated to use for a mathematician, for several reasons:
(1) Because of computability issues that mathematicians don't care about, it currently asks for two maps, and a proof that they do the same thing on overlaps (sounds like a sheaf condition, huh)? Point is that the source of the second map is actually a subset of the source of the first map.
(2) There doesn't seem to be much API for the map that `rec` spits out. In particular, its UMP is missing.